### DISCUSSION TITLE
SHACL and/or ShEx shapes for Linked Places Format

### DISCUSSION THREAD

**Discussion by rybesh** [LinkedPasts/linked-places-format#30] [Ideas]
It would be very useful to have shape validation for LPF. I’ve currently got a bunch of data in LPF v.1.0 that I’d like to bring up to LPF v1.2—would be nice if I could just run a validator over it and see what I need to change. I’ve done a lot of stuff with SHACL so that is something I could work on, but I’d probably need some questions answered that aren’t explicitly addressed in the spec yet.

**Comment by kgeographer** [comment]
FWIW, World Historical Gazetteer (WHG) uses jsonschema.py 3.2.0 on [this schema](https://github.com/WorldHistoricalGazetteer/whgazetteer/blob/main/datasets/static/validate/schema_lpf_v1.2.json) for validating Linked Places format uploads. It has been tweaked a few times over the last few years.

I have not worked with SHACL shapes, assuming I'd have to roll my own validator. The landscape has changed somewhat from 4 years ago, so the approach could/should be revisited at some point.
